Sources/Sentry/SentryANRTrackingIntegration.m

```diff
@@ -15,53 +23,38 @@
 
 #if SENTRY_HAS_UIKIT
 
-/**
- * As we only use the ANR tracking integration for detecting falsely reported OOMs we can use a more
- * defensive value, because we are not reporting any ANRs.
- */
-static NSUInteger const SENTRY_ANR_TRACKER_TIMEOUT_MILLIS = 2000;
-
 @interface
 SentryANRTrackingIntegration ()
 
 @property (nonatomic, strong) SentryANRTracker *tracker;
-@property (nonatomic, strong) SentryAppStateManager *appStateManager;
-@property (nonatomic, strong) SentryCrashWrapper *crashWrapper;
+@property (nonatomic, strong) SentryOptions *options;
 
 @end
 
 @implementation SentryANRTrackingIntegration
 
 - (void)installWithOptions:(SentryOptions *)options
 {
-    SentryDependencyContainer *dependencies = [SentryDependencyContainer sharedInstance];
-    self.crashWrapper = dependencies.crashWrapper;
-
     if ([self shouldBeDisabled:options]) {
         [options removeEnabledIntegration:NSStringFromClass([self class])];
         return;
     }
 
-    self.appStateManager = dependencies.appStateManager;
+    self.tracker = SentryDependencyContainer.sharedInstance.anrTracker;
+    self.tracker.timeoutInterval = options.anrTimeoutInterval;
```

Comment:
`m`: I think we shouldn't change the `timeoutInterval` here. Instead, I think we could pass the `anrTimeoutInterval` to the `[SentryDependencyContainer.sharedInstance anrTracker:options.anrTimeoutInterval`. Otherwise, we change the interval while the anrTracker is already running. What I suggested is also not the cleanest approach, but I think its a bit better.

Reply:
So your suggestions is to change the timeoutInterval only inside DependencyContainer? Because creating a new instance every time this function is called would be a problem, we need to reuse the same tracker for all integrations.

Reply:
No changing it in the `DependencyContainer`, but passing it in as an argument to `[SentryDependencyContainer.sharedInstance anrTracker:options.anrTimeoutInterval]` and then passing it as a constructor argument to `SentryANRTracker`. 
The `DependencyContainer` would only use the argument the first time and then keep a cached version. I know this is not optimal, but it's a pragmatic approach.